pkg/products/threescale/reconciler.go

```diff
@@ -1135,6 +1153,52 @@ func (r *Reconciler) createStsS3Secret(ctx context.Context, serverClient k8sclie
 	return err
 }
 
+func (r *Reconciler) createMCGS3Secret(ctx context.Context, serverClient k8sclient.Client, credSec *corev1.Secret) error {
+	mcgNamespace := r.installation.Spec.NamespacePrefix + mcg.DefaultInstallationNamespace + "-operator"
+	// Retrieve object bucket claim
+	objbc := &noobaav1.ObjectBucketClaim{}
+	err := serverClient.Get(ctx, k8sclient.ObjectKey{Name: mcg.ThreescaleBucketClaim, Namespace: mcgNamespace}, objbc)
+	if err != nil {
+		return fmt.Errorf("failed to get object bucket claim: %w", err)
+	}
+
+	//Retrieve secret with credentials
+	bucketSecret := &corev1.Secret{}
+	err = serverClient.Get(ctx, k8sclient.ObjectKey{Name: mcg.ThreescaleBucketClaim, Namespace: mcgNamespace}, bucketSecret)
+	if err != nil {
+		return fmt.Errorf("failed to get bucket claim secret: %w", err)
+	}
+
+	// Retrieve s3 route
+	s3Route := &routev1.Route{}
+	err = serverClient.Get(ctx, k8sclient.ObjectKey{Name: s3RouteName, Namespace: mcgNamespace}, s3Route)
+	if err != nil {
+		return fmt.Errorf("failed to get s3 route: %w", err)
+	}
+
+	_, err = controllerutil.CreateOrUpdate(ctx, serverClient, credSec, func() error {
+		// Set required fields for MCG
+		for key := range bucketSecret.Data {
+			switch key {
+			case threescaleAmp.AwsAccessKeyID:
+				credSec.Data[threescaleAmp.AwsAccessKeyID] = bucketSecret.Data[threescaleAmp.AwsAccessKeyID]
+			case threescaleAmp.AwsSecretAccessKey:
+				credSec.Data[threescaleAmp.AwsSecretAccessKey] = bucketSecret.Data[threescaleAmp.AwsSecretAccessKey]
+			}
+		}
+		credSec.Data[threescaleAmp.AwsBucket] = []byte(objbc.Spec.BucketName)
+		credSec.Data[threescaleAmp.AwsHostname] = []byte(s3Route.Spec.Host)
+		credSec.Data[threescaleAmp.AwsRegion] = []byte(s3BucketRegion)
+		credSec.Data[threescaleAmp.AwsPathStyle] = []byte(s3PathStyle)
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to create or update blob storage aws credentials secret: %w", err)
```

Comment:
minor. Code looks good , but error message contains `blob storage`, seems copied from another place. This method - createMCGS3Secret(..)

Reply:
Sure - I left it somewhat similar to the original as the threescale secret still contains AWS credentials for blob storage - I'll add MCG into the log to make it clear it's the MCG one.